### PR TITLE
added __floordiv__

### LINKED ISSUE #10110: __floordiv__

### PR DESCRIPTION
**Pull request by rarisenpai**
close #10110

### CODE REVIEW

**Comment by rarisenpai** [comment]
can someone please show why the linting tests are failing?

**Comment by aditya-hari** [comment]
Hey! Thanks for the contribution. The linting test is failing because the newlines haven't been added appropriately, I have flagged them in the review.  In the future, use pre-commit to avoid these issues.

**Comment by rarisenpai** [comment]
Hey thanks for you reply. I did setup precommit I don't know why its not registering probably because I changed in to a new codespace. I also tried adding tests for floordiv and test 1 started failing

**Comment by rarisenpai** [comment]
I also did check the lint seems some of them are from parts of the code I didn't work on like some characters being too long in the backends

**Comment by rarisenpai** [comment]
let me do this again

**Comment by aditya-hari** [comment]
Thanks for the changes! Just need the tests now.

**Comment by rarisenpai** [comment]
Okay will be working on the tests today

**Comment by rarisenpai** [comment]
Hi I need some guidance I added the tests and all of this failed

**Comment by rarisenpai** [comment]
hello

**Review by aditya-hari — APPROVED** [review]

**Comment by rarisenpai** [comment]
okay will do that

**Comment by rarisenpai** [comment]
Hello

**Comment by aditya-hari** [comment]
Hey! Please go through the error log of test-frontend-numpy. It fails with "NameError: name 'pf' not defined".

**Comment by rarisenpai** [comment]
Hi thanks let me fix that

**Comment by rarisenpai** [comment]
Hey I just made the changes you highlighted in test-frontend-numpy and it's
still failing some tests. How should I proceed?

**Comment by rarisenpai** [comment]
There are some tests failing

**Comment by rarisenpai** [comment]
root@codespaces-116095:/workspaces/ivy# pytest ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py::test_numpy_instance_floordiv__
Use Database in ReadOnly Mode with local caching !
======================================================================================================== test session starts ========================================================================================================
platform linux -- Python 3.8.10, pytest-7.1.2, pluggy-1.0.0
rootdir: /workspaces/ivy
plugins: hypothesis-6.55.0, json-report-1.5.0, typeguard-2.13.3, metadata-2.0.4
collected 4 items                                                                                                                                                                                                                   

ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py ...F                                                                                                                                                [100%]

============================================================================================================= FAILURES ==============================================================================================================
___________________________________________________________________________ test_numpy_instance_floordiv__[cpu-ivy.functional.backends.torch-False-False] ___________________________________________________________________________

args = (ivy.array(-1.), ivy.array(-1.)), kwargs = {'out': None}

    @functools.wraps(fn)
    def new_fn(*args, **kwargs):
        """
        Catch all exceptions and raise them in IvyException
    
        Parameters
        ----------
        args
            The arguments to be passed to the function.
    
        kwargs
            The keyword arguments to be passed to the function.
    
        Returns
        -------
            The return of the function, or raise IvyException if error is thrown.
        """
        try:
>           return fn(*args, **kwargs)

ivy/exceptions.py:118: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (ivy.array(-1.), ivy.array(-1.)), kwargs = {'out': None}, cont_fn = <function ContainerWithElementwise.static_floor_divide at 0x7f852b7c8e50>

    @functools.wraps(fn)
    def new_fn(*args, **kwargs):
        """
        Calls `fn` with the *nestable* property of the function correctly handled.
        This means mapping the function to the container leaves if any containers are
        passed in the input.
    
        Parameters
        ----------
        args
            The arguments to be passed to the function.
    
        kwargs
            The keyword arguments to be passed to the function.
    
        Returns
        -------
            The return of the function, with the nestable property handled correctly.
        """
        # if any of the arguments or keyword arguments passed to the function contains
        # a container, get the container's version of the function and call it using
        # the passed arguments.
        if hasattr(ivy.Container, "static_" + fn_name):
            cont_fn = getattr(ivy.Container, "static_" + fn_name)
        else:
            cont_fn = lambda *args, **kwargs: ivy.Container.cont_multi_map_in_function(
                fn, *args, **kwargs
            )
        if ivy.get_nestable_mode() and (
            ivy.nested_any(args, ivy.is_ivy_container, check_nests=True)
            or ivy.nested_any(kwargs, ivy.is_ivy_container, check_nests=True)
        ):
            return cont_fn(*args, **kwargs)
    
        # if the passed arguments does not contain a container, the function using
        # the passed arguments, returning an ivy or a native array.
>       return fn(*args, **kwargs)

ivy/func_wrapper.py:457: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

out = None, args = (ivy.array(-1.), ivy.array(-1.)), kwargs = {}

    @functools.wraps(fn)
    def new_fn(*args, out=None, **kwargs):
        """
        Calls `fn` with the `out` argument handled correctly for performing an inplace
        update.
    
        Parameters
        ----------
        args
            The arguments to be passed to the function.
    
        out
            The array to write the result to.
    
        kwargs
            The keyword arguments to be passed to the function.
    
        Returns
        -------
            The return of the function, with `out` handled correctly for
            inplace updates.
        """
        if out is None:
>           return fn(*args, **kwargs)

ivy/func_wrapper.py:384: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (ivy.array(-1.), ivy.array(-1.)), kwargs = {}, has_out = False, out = None, new_args = (tensor(-1., dtype=torch.float16), tensor(-1., dtype=torch.float16)), new_kwargs = {}

    @functools.wraps(fn)
    def new_fn(*args, **kwargs):
        """
        Converts all `ivy.Array` instances in both the positional and keyword arguments
        into `ivy.NativeArray` instances, and then calls the function with the updated
        arguments.
    
        Parameters
        ----------
        args
            The arguments to be passed to the function.
    
        kwargs
            The keyword arguments to be passed to the function.
    
        Returns
        -------
            The return of the function, with native arrays passed in the arguments.
        """
        if not ivy.get_array_mode():
            return fn(*args, **kwargs)
        # check if kwargs contains an out argument, and if so, remove it
        has_out = False
        out = None
        if "out" in kwargs:
            out = kwargs["out"]
            del kwargs["out"]
            has_out = True
        # convert all arrays in the inputs to ivy.NativeArray instances
        new_args, new_kwargs = ivy.args_to_native(*args, **kwargs)
        # add the original out argument back to the keyword arguments
        if has_out:
            new_kwargs["out"] = out
>       return fn(*new_args, **new_kwargs)

ivy/func_wrapper.py:126: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (tensor(-1., dtype=torch.float16), tensor(-1., dtype=torch.float16)), kwargs = {}

    @functools.wraps(fn)
    def new_fn(*args, **kwargs):
        """
        Calls the function, and then converts all `ivy.NativeArray` instances in
        the function return into `ivy.Array` instances.
    
        Parameters
        ----------
        args
            The arguments to be passed to the function.
    
        kwargs
            The keyword arguments to be passed to the function.
    
        Returns
        -------
            The return of the function, with native arrays as ivy arrays.
        """
        # call unmodified function
>       ret = fn(*args, **kwargs)

ivy/func_wrapper.py:188: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

x1 = tensor(-1., dtype=torch.float16), x2 = tensor(-1., dtype=torch.float16)

    @with_unsupported_dtypes({"1.11.0 and below": ("float16", "complex")}, backend_version)
    def floor_divide(
        x1: Union[float, torch.Tensor],
        x2: Union[float, torch.Tensor],
        /,
        *,
        out: Optional[torch.Tensor] = None,
    ) -> torch.Tensor:
        x1, x2 = ivy.promote_types_of_inputs(x1, x2)
        if ivy.exists(out):
            if not ivy.is_float_dtype(out):
                return ivy.inplace_update(
                    out, torch.floor(torch.div(x1, x2)).type(out.dtype)
                )
>       return torch.floor(torch.div(x1, x2), out=out).type(x1.dtype)
E       RuntimeError: "floor" "_vml_cpu" not implemented for 'Half'

ivy/functional/backends/torch/elementwise.py:435: RuntimeError

During handling of the above exception, another exception occurred:

frontend = 'numpy'

    @handle_frontend_method(
>       class_tree=CLASS_TREE,
        init_tree="numpy.array",
        method_name="__floordiv__",
        dtype_and_x=helpers.dtype_and_values(
            available_dtypes=helpers.get_dtypes("numeric"),
            num_arrays=2,
            large_abs_safety_factor=4,
            shared_dtype=True,
            safety_factor_scale="linear",
        ),
    )

ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:1219: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:1241: in test_numpy_instance_floordiv__
    helpers.test_frontend_method(
ivy_tests/test_ivy/helpers/function_testing.py:1521: in test_frontend_method
    ret, ret_np_flat = get_ret_and_flattened_np_array(
ivy_tests/test_ivy/helpers/function_testing.py:1803: in get_ret_and_flattened_np_array
    ret = fn(*args, **kwargs)
ivy/functional/frontends/numpy/ndarray/ndarray.py:294: in __floordiv__
    return np_frontend.floor_divide(self._ivy_array, value)
ivy/functional/frontends/numpy/ufunc/methods.py:260: in __call__
    return self.func(*args, **kwargs)
ivy/functional/frontends/numpy/func_wrapper.py:398: in new_fn
    return fn(*args, out=out, **kwargs)
ivy/functional/frontends/numpy/func_wrapper.py:33: in new_fn
    return fn(*args, dtype=np_frontend.to_ivy_dtype(dtype), **kwargs)
ivy/functional/frontends/numpy/func_wrapper.py:284: in new_fn
    ret = fn(*args, order=order, **kwargs)
ivy/functional/frontends/numpy/func_wrapper.py:257: in new_fn
    return fn(*ivy_args, **ivy_kwargs)
ivy/functional/frontends/numpy/func_wrapper.py:119: in new_fn
    return fn(*args, **kwargs)
ivy/functional/frontends/numpy/func_wrapper.py:341: in new_fn
    ret = fn(*args, **kwargs)
ivy/functional/frontends/numpy/mathematical_functions/arithmetic_operations.py:238: in _floor_divide
    ret = ivy.floor_divide(x1, x2, out=out)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = (ivy.array(-1.), ivy.array(-1.)), kwargs = {'out': None}

    @functools.wraps(fn)
    def new_fn(*args, **kwargs):
        """
        Catch all exceptions and raise them in IvyException
    
        Parameters
        ----------
        args
            The arguments to be passed to the function.
    
        kwargs
            The keyword arguments to be passed to the function.
    
        Returns
        -------
            The return of the function, or raise IvyException if error is thrown.
        """
        try:
            return fn(*args, **kwargs)
        except (IndexError, ValueError, AttributeError) as e:
            _print_traceback_history()
            raise ivy.exceptions.IvyError(fn.__name__, str(e))
        except Exception as e:
            _print_traceback_history()
>           raise ivy.exceptions.IvyBackendException(fn.__name__, str(e))
E           ivy.exceptions.IvyBackendException: torch: floor_divide: "floor" "_vml_cpu" not implemented for 'Half'
E           Falsifying example: test_numpy_instance_floordiv__(
E               dtype_and_x=(['float16', 'float16'],
E                [array(-1., dtype=float16), array(-1., dtype=float16)]),
E               init_flags=num_positional_args=0. native_arrays=[False]. as_variable=[False]. ,
E               frontend_method_data=FrontendMethodData(ivy_init_module=<module 'ivy.functional.frontends.numpy' from '/workspaces/ivy/ivy/functional/frontends/numpy/__init__.py'>, framework_init_module=<module 'numpy' from '/usr/local/lib/python3.8/dist-packages/numpy/__init__.py'>, init_name='array', method_name='__floordiv__'),
E               method_flags=num_positional_args=1. native_arrays=[False]. as_variable=[False]. ,
E               frontend='numpy',
E           )
E           
E           You can reproduce this example by temporarily adding @reproduce_failure('6.55.0', b'AXicY2AAAkYGCGBEY4MBAABuAAU=') as a decorator on your test case

ivy/exceptions.py:124: IvyBackendException
------------------------------------------------------------------------------------------------------- Captured stdout call --------------------------------------------------------------------------------------------------------
  File "/workspaces/ivy/ivy/exceptions.py", line 118, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 457, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 384, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 126, in new_fn
    return fn(*new_args, **new_kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 188, in new_fn
    ret = fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/functional/backends/torch/elementwise.py", line 435, in floor_divide
    return torch.floor(torch.div(x1, x2), out=out).type(x1.dtype)

During the handling of the above exception, another exception occurred:

  File "/workspaces/ivy/ivy/exceptions.py", line 118, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 457, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 384, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 126, in new_fn
    return fn(*new_args, **new_kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 188, in new_fn
    ret = fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/functional/backends/torch/elementwise.py", line 435, in floor_divide
    return torch.floor(torch.div(x1, x2), out=out).type(x1.dtype)

During the handling of the above exception, another exception occurred:

  File "/workspaces/ivy/ivy/exceptions.py", line 118, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 457, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 384, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 126, in new_fn
    return fn(*new_args, **new_kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 188, in new_fn
    ret = fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/functional/backends/torch/elementwise.py", line 435, in floor_divide
    return torch.floor(torch.div(x1, x2), out=out).type(x1.dtype)

During the handling of the above exception, another exception occurred:

  File "/workspaces/ivy/ivy/exceptions.py", line 118, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 457, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 384, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 126, in new_fn
    return fn(*new_args, **new_kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 188, in new_fn
    ret = fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/functional/backends/torch/elementwise.py", line 435, in floor_divide
    return torch.floor(torch.div(x1, x2), out=out).type(x1.dtype)

During the handling of the above exception, another exception occurred:

  File "/workspaces/ivy/ivy/exceptions.py", line 118, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 457, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 384, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 126, in new_fn
    return fn(*new_args, **new_kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 188, in new_fn
    ret = fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/functional/backends/torch/elementwise.py", line 435, in floor_divide
    return torch.floor(torch.div(x1, x2), out=out).type(x1.dtype)

During the handling of the above exception, another exception occurred:

  File "/workspaces/ivy/ivy/exceptions.py", line 118, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 457, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 384, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 126, in new_fn
    return fn(*new_args, **new_kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 188, in new_fn
    ret = fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/functional/backends/torch/elementwise.py", line 435, in floor_divide
    return torch.floor(torch.div(x1, x2), out=out).type(x1.dtype)

During the handling of the above exception, another exception occurred:

  File "/workspaces/ivy/ivy/exceptions.py", line 118, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 457, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 384, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 126, in new_fn
    return fn(*new_args, **new_kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 188, in new_fn
    ret = fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/functional/backends/torch/elementwise.py", line 435, in floor_divide
    return torch.floor(torch.div(x1, x2), out=out).type(x1.dtype)

During the handling of the above exception, another exception occurred:

  File "/workspaces/ivy/ivy/exceptions.py", line 118, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 457, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 384, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 126, in new_fn
    return fn(*new_args, **new_kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 188, in new_fn
    ret = fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/functional/backends/torch/elementwise.py", line 435, in floor_divide
    return torch.floor(torch.div(x1, x2), out=out).type(x1.dtype)

During the handling of the above exception, another exception occurred:

  File "/workspaces/ivy/ivy/exceptions.py", line 118, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 457, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 384, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 126, in new_fn
    return fn(*new_args, **new_kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 188, in new_fn
    ret = fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/functional/backends/torch/elementwise.py", line 435, in floor_divide
    return torch.floor(torch.div(x1, x2), out=out).type(x1.dtype)

During the handling of the above exception, another exception occurred:

  File "/workspaces/ivy/ivy/exceptions.py", line 118, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 457, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 384, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 126, in new_fn
    return fn(*new_args, **new_kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 188, in new_fn
    ret = fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/functional/backends/torch/elementwise.py", line 435, in floor_divide
    return torch.floor(torch.div(x1, x2), out=out).type(x1.dtype)

During the handling of the above exception, another exception occurred:

  File "/workspaces/ivy/ivy/exceptions.py", line 118, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 457, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 384, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 126, in new_fn
    return fn(*new_args, **new_kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 188, in new_fn
    ret = fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/functional/backends/torch/elementwise.py", line 435, in floor_divide
    return torch.floor(torch.div(x1, x2), out=out).type(x1.dtype)

During the handling of the above exception, another exception occurred:

  File "/workspaces/ivy/ivy/exceptions.py", line 118, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 457, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 384, in new_fn
    return fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 126, in new_fn
    return fn(*new_args, **new_kwargs)
  File "/workspaces/ivy/ivy/func_wrapper.py", line 188, in new_fn
    ret = fn(*args, **kwargs)
  File "/workspaces/ivy/ivy/functional/backends/torch/elementwise.py", line 435, in floor_divide
    return torch.floor(torch.div(x1, x2), out=out).type(x1.dtype)

During the handling of the above exception, another exception occurred:

========================================================================================================= warnings summary ==========================================================================================================
../../usr/local/lib/python3.8/dist-packages/flatbuffers/compat.py:19
  /usr/local/lib/python3.8/dist-packages/flatbuffers/compat.py:19: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py::test_numpy_instance_floordiv__[cpu-ivy.functional.backends.jax-False-False]
  /workspaces/ivy/ivy_tests/test_ivy/helpers/function_testing.py:1573: RuntimeWarning: overflow encountered in floor_divide
    frontend_ret = ins_gt.__getattribute__(frontend_method_data.method_name)(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
====================================================================================================== short test summary info ======================================================================================================
FAILED ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py::test_numpy_instance_floordiv__[cpu-ivy.functional.backends.torch-False-False] - ivy.exceptions.IvyBackendException: torch: floor_divide: "floor"...
============================================================================================= 1 failed, 3 passed, 2 warnings in 18.11s ==============================================================================================
root@codespaces-116095:/workspaces/ivy#

**Comment by rarisenpai** [comment]
This is the error message I am getting after running the test_ndarray function for __floordiv__. Its passing 3 tests and failing one

File: ivy/functional/frontends/numpy/ndarray/ndarray.py [hunk 1]
```diff
@@ -290,6 +290,9 @@ def __mul__(self, value, /):
     def __truediv__(self, value, /):
         return np_frontend.true_divide(self._ivy_array, value)
 
+    def __floordiv__(self, value, /):
+        return ivy.floor_divide(self._ivy_array,value)
```
Comment (Daniel4078) [inline ivy/functional/frontends/numpy/ndarray/ndarray.py:294]: small lint error here: ivy/functional/frontends/numpy/ndarray/ndarray.py:294:48: E231 missing whitespace after ','

**Comment by aditya-hari** [comment]
Hey so besides the minor linting errors, here's what you need to for the tests -  
 
1. The assume statements about zero seem to deal with the zero division errors, so that should be clear. If you run into further problems though, then just explicitly generate non-zero denominators.  
2. E RuntimeError: "floor" "_vml_cpu" not implemented for 'Half' means that floor divide doesn't work for float16 in torch. Just disallow that data type, and any other ones that don't work

**Review by aditya-hari — CHANGES_REQUESTED** [review]
This should be np_frontend.floor_divide

**Comment by rarisenpai** [comment]
hi made those changes

**Comment by rarisenpai** [comment]
I also made the changes to the tests 
Kindly offer any other guidance in case I still need to make more changes

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py [hunk 2]
```diff
@@ -1215,6 +1215,50 @@ def test_numpy_instance_mul__(
...
+)
+def test_numpy_instance_floordiv__(
+    dtype_and_x,
+    frontend_method_data,
+    init_flags,
+    method_flags,
+    frontend,
+):
+    input_dtypes, xs = dtype_and_x
+    # Make sure it's not dividing value too close to zero
```
Comment (aditya-hari) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:1240]: Don't think the comments are necessary.

**Comment by aditya-hari** [comment]
Looks good to me! Thanks for the contribution!

**Pull request merged**